in_toto/models/statement.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python
+
+# Copyright New York University and the in-toto contributors
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+<Program Name>
+  statement.py
+
+<Author>
+  Chasen Bettinger <bettingerchasen@gmail.com>
+
+<Started>
+  Mar 25, 2023
+
+<Copyright>
+  See LICENSE for licensing information.
+
+<Purpose>
+  Provides a class that represents a Statement, the middle layer of an 
+  attestation, binding it to a particular subject and unambiguously 
+  identifying the types of the Predicate as defined in the in-toto 
+  attestion spec.
+"""
+
+import attr
+import securesystemslib.formats
+from in_toto.models.common import Signable
+
+STATEMENT_TYPE = "https://in-toto.io/Statement/v1"
+FILENAME_FORMAT = "{step_name}.{keyid:.8}.link"
```

Comment:
I think these don't belong at the statement layer. We'll have to handle them smartly for the link predicate only.